autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/persister/CredentialsLastModifiedWinsStrategy.kt

```diff
@@ -37,27 +37,43 @@ class CredentialsLastModifiedWinsStrategy(
         clientModifiedSince: String,
     ): SyncMergeResult {
         Timber.d("Sync-autofill-Persist: ======= MERGING TIMESTAMP =======")
+        var hasDataChangedWhileSyncing = false
         return kotlin.runCatching {
             runBlocking(dispatchers.io()) {
                 credentials.entries.forEach { entry ->
                     val localCredential = credentialsSync.getCredentialWithSyncId(entry.id)
                     if (localCredential == null) {
                         processNewEntry(entry, clientModifiedSince)
                     } else {
-                        processExistingEntry(localCredential, entry, clientModifiedSince)
+                        val localId = localCredential.id!!
+                        if (entry.isDeleted()) {
+                            Timber.d("Sync-autofill-Persist: >>> delete local ${localCredential.id}")
+                            credentialsSync.deleteCredential(localId)
```

Comment:
We should `return` at this point. If entry deleted, rest of the logic should not be executed.

Reply:
discussed offline

Reply:
correct, adding the change now.